File: Eugle.php
<?php

function isEugle($g) {
    $h = array();
    $e = array();
    $idk = 0;
    foreach ($g as $i) {
        array_push($h, array_sum($i));
    }
    foreach ($h as $j) {
        array_push($e, $j % 2);
    }
    foreach ($e as $k) {
        if ($k == 1) {
            $idk += 1;
            if ($idk == 3) {
                return false;
            }
        }
    }
    return true;
}

echo "Enter nums: \n";
$g = array();
while (true) {
    $row = trim(fgets(STDIN));
    if ($row == ".") {
        break;
    }
    $row_nums = explode(",", $row);
    foreach ($row_nums as $num) {
        if (!is_numeric($num)) {
            echo "Please enter integers only\n";
            exit();
        }
    }
    array_push($g, $row_nums);
}

echo "It is Eugle: " . var_export(isEugle($g), true) . "\n";

?>

File: Eugle.py
import sys

#######################################################
def isEugle(g):
    h = []
    e = []
    idk = 0
    for i in g:
        h.append(sum(i))
    for j in h:
        e.append(j%2)
    for k in e:
        if k == 1:
            idk += 1
            if idk == 3:
                return False
    return True
#######################################################

#######################################################
def pathExists():
    return
#######################################################

#######################################################
class FIFO():
    def __init__(self):
        self.queue1 = []
        self.queue2 = []

    def push(self, value):
        self.queue1.append(value)

    def pop(self):
        return self.queue1.pop(0)
#######################################################

#######################################################
print("Enter nums, seperate them by ',' and press 'ENTER' if you want to end one array, press '.' to end it all: ")
g = []
while True:
    row = input().strip()
    if row == ".":
        break
    row_nums = row.split(",")
    try:
        row_nums = [int(num) for num in row_nums]
    except ValueError:
        print("Please enter integers only")
        sys.exit()
    g.append(row_nums)

print(f"It is Eugle: {isEugle(g)}")
#######################################################
